bubble_sort.rb

```diff
@@ -0,0 +1,56 @@
+# FILE: bubble_sort.rb
+# Programmer: Steven Anderson
+# Last Modified: 8 November 2019
+# Description: A sorting algorithm that compares, from left to right, each element of the array and swaps the items so that
+#              smallest goes to largest. Based on a fimiliarity with many algorithms in the Java programming language, this is 
+#              a Ruby implementation of the, ever useful, Bubble Sort algorithm.
+#
+#              Importantly, this works for an array of any data type. For strings, capitalized letters come first then lower case.
+#              Ruby treats comaprisons between letters by their ascii values. Meaning A = 65, B = 66 etc. and a = 97, b = 98 etc.
+#              More can be read here about how Ruby comapres strings: http://www.evc-cit.info/cit020/beginning-programming/chp_02/string_compare.html
```

Comment:
I love the description and organization efforts!

For the purposes of trying to keep the code base readable as it gets bigger please remove this before creating a pull request. Thankfully GitHub tracks everything you contribute! 👍

Reply:
- I will absolutely drop that in future pull requests, thanks for the input! You get used to a certain way of documenting stuf, and it's a habit of mine from college.
- Towards your initial comment regarding putting capitalized letters first. The reason this is like this, is because the default way to compare characters/strings is based on the ascii values of each character. Since the bubble will move the elements in to ascending order it will sort the smaller, capital letter values to the front of the array. I can probably find a way to alternate that if it makes more sense to keep the capitalized letter last?
- As far as feature requests, I will be adding those features to my code tomorrow, as I will have dedicated time for this. 

**Discussion points:**
My first instinct is to create a class that just includes each of the methods that we are writing for each *way* to sort. This would mean that we would basically need to just copy and paste the sorting methods in to the class and use dot-notation to access them. That said, I am a little iffy doing that as that makes actually using those methods a little un-intuitive. I'm going to do some research and play around with some options to see the best way we can clean it all up and make it all easier to use. I think your look makes a lot of sense but it will definitely be on us, as the programmers, to make sure we use meaningful instance naming. So, having a class that is called say, ``Sort`` would work nicely then include the sorting algs that were written so far (and the future ones) within that class. After that, calling those algs will come down to good naming convention. ``bubble_sort = Sort.new`` followed by ``bubble_sort.bubble_sort(name_of_the_array)`` will give us a lot more power outside of the class, while maintaining integrity of the class with basically no risk.   
   
So, I will respond to what you mentioned directly:
- So far we for sure have a template for a ``Sort`` class. The common methods will be all of the sorting algs we have so far. We might be able to work in a class that does ``WordProcessing`` or something to handle user input, but I think that may be better implemented right now strictly as a method, as it currently is.
- After looking some more at modules, I think that actually may be a really great way to implement these algorithms, since most of these algs, while taking in arrays mostly don't have a lot of the same things that will be within a regular class, though a class might need to implement them. So on second thought, I think maybe a ``Sort`` module may be the best way to implement these algs. That said, I think that keeping that module on a separate file is going to be the cleanest way to write it. Any other file that wants to access the module will need to use the `require_relative "name_of_file_without_the_extension"`
             `include NameOfTheModule` at the top of the file that wants to implement it.

Reply:
Actually, I think I just realized what you meant with how it is sorting. Are you asking about getting the sort to sort each of the words on their alphabetical value, regardless of if it is capitalized or not?

Reply:
Yes, your last comment is exactly what I intended to convey for sorting order. The goal is to have the order be alphabetical regardless of capitalized words being in the sort; using `.downcase` is great for this. There are additional steps to create the desired output of the sorted array in alphabetical order, including the capitalized words. The output display should also show the words in their original capitalized state. For example, using the words array: `"Senate"` would display in its correct dictionary alphabetical order, and also display as capitalized. View the output of running `sort(words)` for an example of this.

I'm new to giving feedback for code reviews and discussions, and will work to improve the clarity of my comments. I appreciate you practicing this process with me, and any feedback is welcome! Thanks for understanding while I learn 😄

Reply:
Yeah, we're all learning! I just misunderstood what was being talked about. I should be able to make that change to the sort. I'll be working on some of this stuff hopefully today. Thanks for the feedback!